addons/product/models/product_attribute.py

```diff
@@ -22,8 +22,30 @@ class ProductAttribute(models.Model):
         ('always', 'Always'),
         ('dynamic', 'Only when the product is added to a sales order')],
         default='always',
-        string="Create Variants",
-        help="Check this if you want to create multiple variants for this attribute.", required=True)
+        states={'used_on_product': [('readonly', True)]},
+        string="Variants Creation Mode",
+        help="This cannot be changed if the attribute is used on at least one product.", required=True)
+
+    state = fields.Selection(compute='_compute_state',
+        selection=[
+            ('not_used_on_product', "Not used on Product"),
+            ('used_on_product', "Used on Product")
+        ],
+        default='not_used_on_product',
+    )
+    product_tmpl_ids = fields.Many2many('product.template', string="Related Products", compute='_compute_products', store=True)
+
+    @api.multi
```

Comment:
@api.multi is deprecated in master https://github.com/odoo/odoo/pull/34775

Reply:
Thanks for the notice! We didn't get the message as it's not merged yet.